### DISCUSSION TITLE
How to install two formulas with the same name?

### DISCUSSION THREAD

**Discussion by kud** [Homebrew/discussions#4782] [Everyday usage]
### Description of issue

Hello Homebrew community,

I recently stumbled upon two projects which are both available via Homebrew:

- [getcord/spr](https://github.com/getcord/spr)
- [ejoffe/spr](https://github.com/ejoffe/spr)

The issue is that both projects have the same command name spr. I was wondering how I could install both of them and perhaps rename one of the installed binaries to spr2.

I considered an approach where I install one, unlink it, install the other, and then link the first one again. After this, I thought about creating a symlink for the second one with the name spr2. But I'm wondering if there's a more elegant or recommended way to achieve this via Homebrew.

Any guidance would be much appreciated!

Thank you!

**Comment by gromgit** [comment]
> The issue is that both projects have the same command name spr.

You sure about that? `ejoffe`'s stuff seems to be run as `git` subcommands, and `git XYZ` maps to an executable named `git-spr`. That tallies with the release tarballs:
```bash
$ tar tvf ~/Downloads/spr_linux_arm64.tar.gz 
-rw-r--r-- runner/staff   1072 2023-09-08 07:59 LICENSE
-rw-r--r-- runner/staff  10609 2023-09-08 07:59 readme.md
-rwxr-xr-x runner/staff 8323072 2023-09-08 08:00 git-spr
-rwxr-xr-x runner/staff 6356992 2023-09-08 08:01 git-amend
-rwxr-xr-x runner/staff 3473408 2023-09-08 08:01 spr_reword_helper
```
so I'm not seeing a name clash here.

In the general case, though, when you have multiple third-party formulae with conflicting file names, there's no elegant solution: either someone has to rename their files, or you'll have to do the unlink two-step you described.

**Reply by kud** [reply]
At first, I thought the same like you, one is `spr` and the other one is `git spr`. But:

<img width="1599" alt="screenshot-2023-09-12-16-09-25@2x" src="https://github.com/Homebrew/discussions/assets/655838/c0b42ad4-63a2-41af-91d8-480211a4ca7e">

🤷

**Reply by gromgit** [reply]
Sorry, running on fumes right now. Yeah, this is conflicting _formula_ names, which is worse--you can't even unpack the second instance in the Homebrew Cellar directory. One party will have to rename...or you'll have to fork the tap and do it yourself, but that commits you to maintaining said fork going forward.

**Reply by kud** [reply]
Let' s see how it goes:

https://github.com/ejoffe/spr/issues/352
https://github.com/getcord/spr/issues/177

**Reply by davinkevin** [reply]
Same projects, Same issues…  😔

**Reply by gromgit** [reply]
> Same projects, Same issues… 😔

But slightly different answer now. Since the spacedentist version is now in Homebrew core, and ejoffe has declined to rename their formula, someone will have to create and maintain their own tap with a renamed formula, for ejoffe's project.

**Reply by kud** [reply]
Lots of fun just for a name. 🎉

**Reply by gromgit** [reply]
There are only two hard problems in computer science: cache invalidation and naming things. --Phil Karlton

**Reply by kud** [reply]
Oh god, yes! This.
